activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/main/java/org/activiti/cloud/services/modeling/validation/extensions/ExtensionsJsonSchemaValidator.java

```diff
@@ -31,43 +32,52 @@
 /**
  * The generic JSON extensions schema for all the models
  */
-public abstract class ExtensionsJsonSchemaValidator extends JsonSchemaModelValidator implements ModelExtensionsValidator {
+public abstract class ExtensionsJsonSchemaValidator extends JsonSchemaModelValidator implements
+    ModelExtensionsValidator {
 
-    @Autowired
-    private JsonConverter<Model> extensionsConverter;
+  @Autowired
+  private JsonConverter<Model> extensionsConverter;
 
-    @Override
-    public void validate(byte[] bytes,
-                         ValidationContext validationContext) {
-        super.validate(bytes,
-                       validationContext);
+  @Override
+  public void validate(byte[] bytes,
+                       ValidationContext validationContext) {
+    super.validate(bytes,
+                   validationContext);
 
-        if (!validationContext.isEmpty()) {
-            validateExtensionsInContext(bytes,
-                                         validationContext);
-        }
+    if (!validationContext.isEmpty()) {
+      validateExtensionsInContext(bytes,
+                                  validationContext);
     }
+  }
+
+  @Override
+  public Collection<ModelValidationError> validateAndReturnErrors(byte[] modelContent,
+                                                                  ValidationContext validationContext) {
+    return getValidationErrors(convertBytesToModel(modelContent),
+                               validationContext);
+  }
 
-    private void validateExtensionsInContext(byte[] bytes,
-                                              ValidationContext validationContext) {
-        List<ModelValidationError> validationExceptions = getValidationErrors(convertBytesToModel(bytes),
-                                                                              validationContext);
-        if (!validationExceptions.isEmpty()) {
-            throw new SemanticModelValidationException("Semantic model validation errors encountered: "
-                    + validationExceptions.stream().map(ModelValidationError::getDescription).collect(Collectors.joining(",")),
-                                                       validationExceptions);
-        }
+  private void validateExtensionsInContext(byte[] bytes,
+                                           ValidationContext validationContext) {
+    List<ModelValidationError> validationExceptions = getValidationErrors(convertBytesToModel(bytes),
+                                                                          validationContext);
+    if (!validationExceptions.isEmpty()) {
+      throw new SemanticModelValidationException("Semantic model validation errors encountered: "
```

Comment:
We should probably transform this exception to a collection of `SemanticModelError`, if we are going to create different error classes as said above.

Reply:
Good idea